src/types/machine_config.go

```diff
@@ -0,0 +1,46 @@
+package types
+
+import "time"
+
+type Labels struct {
+	Role string `yaml:"machineconfiguration.openshift.io/role"`
+}
+
+type Metadata struct {
+	CreationTimestamp *time.Time `yaml:"creationTimestamp"`
+	Labels            Labels     `yaml:"labels"`
+	Name              string     `yaml:"name"`
+}
+
+type Ignition struct {
+	Version string `yaml:"version"`
+}
+
+type User struct {
+	Name              string   `yaml:"name"`
+	SshAuthorizedKeys []string `yaml:"sshAuthorizedKeys"`
+}
+
+type Passwd struct {
+	Users []User `yaml:"users"`
+}
+
+type Config struct {
+	Ignition Ignition `yaml:"ignition"`
+	Passwd   Passwd   `yaml:"passwd"`
+}
+
+type Spec struct {
+	Config          Config  `yaml:"config"`
+	Fips            bool    `yaml:"fips"`
+	KernelArguments *string `yaml:"kernelArguments"`
+	KernelType      *string `yaml:"kernelType"`
+	OsImageURL      *string `yaml:"osImageURL"`
+}
+
+type MachineConfig struct {
```

Comment:
I think that you can import it from bootkube instead of generating it by yourself

Reply:
Bootkube start the cluster-bootstrap-operator that just apply these yamls./
This yaml is generated by the openshift-installerm which isn't a nice dependency to have.

Reply:
I will need to check that.

Reply:
Should be able to get this from here https://github.com/openshift/machine-config-operator/blob/master/pkg/apis/machineconfiguration.openshift.io/v1/types.go

Reply:
Will do it it a different task, https://issues.redhat.com/browse/MGMT-2847

Reply:
Why? This should be pretty easy I think, right?

Reply:
Because we wanted to insert this task yesterday at 22:00, obviously that didn't happened so I will do it in the same PR :)

Reply:
obsolete